url47087446e9_kellrott_SparqlScape_tgz-pJ8-org_json_TestJ8/src/org/sparql/Client.java

```diff
@@ -0,0 +1,104 @@
+package org.sparql;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONArray;
+
+public class Client {
+
+	private URL endpoint;
+
+	public Client(URL endpoint) {
+		this.endpoint = endpoint;
+	}
+
+	public Results select(String query) {
+		Results outList = null;
+
+		try {
+			String queryURL = endpoint.toString() + "?output=json&query=" + URLEncoder.encode( query, "UTF8" );
+			System.out.println(queryURL);
+			URLConnection conn = (new URL(queryURL)).openConnection();
+
+			InputStream in = conn.getInputStream();
+			BufferedReader br = new BufferedReader(new InputStreamReader(in));
+			StringBuilder sb = new StringBuilder();
+
+			String line = null;
+			do { 
+				line = br.readLine();
+				if ( line != null )
+					sb.append(line);			
+			} while (line != null);					
+			JSONObject result = new JSONObject( sb.toString() );			
+
+			List<String> varList = new ArrayList<String>();
+			JSONArray vars = result.getJSONObject("head").getJSONArray("vars");
+			for ( int i = 0; i < vars.length(); i++) {
+				varList.add( vars.get(i).toString()  );
+			}
+			outList = new Results(varList);
+			//System.err.println(varList);
+			JSONArray bind = result.getJSONObject("results").getJSONArray("bindings");
+			for ( int i = 0; i < bind.length(); i++) {
+				Row row = new Row();
+				JSONObject sparqlRow = bind.getJSONObject(i);
+				for ( String key : varList ) {
+					if ( sparqlRow.has(key)) {
+						JSONObject r = sparqlRow.getJSONObject(key);
+						if (r.has("type")) {
+							String type = r.getString("type"); 
+							if ( type.compareTo("uri") == 0 ) {
+								row.put(key, new SparqlData( r.getString("value"), true ) );
+							} else if (type.compareTo("literal") == 0 ) {
+								row.put(key, new SparqlData( r.getString("value"), false ) );						
+							}
+						}
+					}
+				}
+				outList.add(row);
+			}
+			//System.err.print(result);
+		} catch (UnsupportedEncodingException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		} catch (MalformedURLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This code contains a potential information leakage in the handling of the following *exception logging*: (MalformedURLException e). Stack trace information reveals variable names, method names, line numbers, and other parts of the code.

**Fix**
Release data more selectively, or declassify the data before it&#39;s released. 

**More info**
[Learn more about information leakage](https://cwe.mitre.org/data/definitions/497.html) (external link).